src/Drivers/Panapal/Panapal.php

```diff
@@ -0,0 +1,236 @@
+<?php
+
+namespace Shetabit\Multipay\Drivers\Panapal;
+
+use GuzzleHttp\Client;
+use Shetabit\Multipay\Abstracts\Driver;
+use Shetabit\Multipay\Exceptions\InvalidPaymentException;
+use Shetabit\Multipay\Exceptions\PurchaseFailedException;
+use Shetabit\Multipay\Contracts\ReceiptInterface;
+use Shetabit\Multipay\Invoice;
+use Shetabit\Multipay\Receipt;
+use Shetabit\Multipay\RedirectionForm;
+use Shetabit\Multipay\Request;
+
+class Panapal extends Driver
+{
+    /**
+     * Panapal Client.
+     *
+     * @var object
+     */
+    protected \GuzzleHttp\Client $client;
+
+    /**
+     * Invoice
+     *
+     * @var Invoice
+     */
+    protected $invoice;
+
+    /**
+     * Driver settings
+     *
+     * @var object
+     */
+    protected $settings;
+
+    /**
+     * Panapal constructor.
+     * Construct the class with the relevant settings.
+     *
+     * @param $settings
+     */
+    public function __construct(Invoice $invoice, $settings)
+    {
+        $this->invoice($invoice);
+        $this->settings = (object) $settings;
+        $this->client = new Client();
+    }
+
+    /**
+     * Purchase Invoice.
+     *
+     * @return string
+     *
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function purchase()
+    {
+        $details = $this->invoice->getDetails();
+
+        $name = '';
+        if (!empty($details['name'])) {
+            $name = $details['name'];
+        }
+
+        $mobile = '';
+        if (!empty($details['mobile'])) {
+            $mobile = $details['mobile'];
+        } elseif (!empty($details['phone'])) {
+            $mobile = $details['phone'];
+        }
+
+        $email = '';
+        if (!empty($details['mail'])) {
+            $email = $details['mail'];
+        } elseif (!empty($details['email'])) {
+            $email = $details['email'];
+        }
+
+        $desc = '';
+        if (!empty($details['desc'])) {
+            $desc = $details['desc'];
+        } elseif (!empty($details['description'])) {
+            $desc = $details['description'];
+        }
+
+        $amount = $this->invoice->getAmount();
+        if ($this->settings->currency != 'T') {
+            $amount /= 10;
+        }
+        $amount = intval(ceil($amount));
+
+        $orderId = crc32($this->invoice->getUuid());
```

Comment:
should order id be integer ? the order id is unique between all orders of all users or uqique for order of each user ?
I suggest not using  crc32 because it has low cardinality and can cause duplicate orderId.